tests/integration/utils.py

```diff
@@ -251,17 +251,19 @@ def fetch_deps_and_check_output(
         _validate_expected_dep_file_contents(deps_content_file, Path(output_folder))
 
     if test_params.check_deps_checksums:
-        files_checksums = _calculate_files_checksums_in_dir(os.path.join(output_folder, "deps"))
+        files_checksums = _calculate_files_checksums_in_dir(output_folder.joinpath("deps"))
         expected_files_checksums = _load_json(
-            os.path.join(test_data_dir, test_case, "fetch_deps_sha256sums.json")
+            test_data_dir.joinpath(test_data_dir)
+            .joinpath(test_case)
+            .joinpath("fetch_deps_sha256sums.json")
```

Comment:
I found out that joinpath takes multiple arguments, so this could be

```suggestion
            test_data_dir.joinpath(test_case, "fetch_deps_sha256sums.json")
```
IMO that's a bit more readable, what do you think? Maybe we could use this in other places where we currently use multiple joinpaths?

Or, the other option would be this
```suggestion
            test_data_dir / test_case / "fetch_deps_sha256sums.json"
```